esp/services/ws_access/ws_accessService.cpp

```diff
@@ -2162,6 +2162,572 @@ bool Cws_accessEx::onResourcePermissions(IEspContext &context, IEspResourcePermi
 
     return true;
 }
+    
+bool Cws_accessEx::onQueryViews(IEspContext &context, IEspQueryViewsRequest &req, IEspQueryViewsResponse &resp)
+{
+    try
```

Comment:
Versions are automatically handled by esp framework. Automatically generated ws_access.esp file contains code to check for the client version and throw appropriate error:
```
[build/generated/ws_access.esp]:
if (clientVer!=-1.0 && clientVer<1.1)
    throw MakeStringException(-1, "Client version is too old, please update your client application.");
```